quarkus-caffeine should be part of the camel-core dependencies

Otherwise, you end up with the following error when executing a native image:
```
2019-03-25 12:31:29,479 INFO  [io.qua.cam.cor.run.CamelRuntime] (main) Apache Camel  (CamelContext: camel-1) is starting
java.lang.IllegalStateException: com.github.benmanes.caffeine.cache.SILMS
	at com.github.benmanes.caffeine.cache.LocalCacheFactory.newBoundedLocalCache(Unknown Source)
	at com.github.benmanes.caffeine.cache.BoundedLocalCache$BoundedLocalManualCache.<init>(BoundedLocalCache.java:3043)
	at com.github.benmanes.caffeine.cache.BoundedLocalCache$BoundedLocalManualCache.<init>(BoundedLocalCache.java:3039)
	at com.github.benmanes.caffeine.cache.Caffeine.build(Caffeine.java:926)
	at org.apache.camel.support.LRUCache.<init>(LRUCache.java:145)
	at org.apache.camel.support.LRUCache.<init>(LRUCache.java:113)
	at org.apache.camel.support.LRUSoftCache.<init>(LRUSoftCache.java:67)
	at org.apache.camel.support.LRUSoftCache.<init>(LRUSoftCache.java:63)
	at org.apache.camel.support.LRUSoftCache.<init>(LRUSoftCache.java:59)
	at org.apache.camel.support.LRUCacheFactory.newLRUSoftCache(LRUCacheFactory.java:156)
	at org.apache.camel.component.properties.PropertiesComponent.<init>(PropertiesComponent.java:69)
	at io.quarkus.camel.core.runtime.CamelRuntime.createPropertiesComponent(CamelRuntime.java:154)
	at io.quarkus.camel.core.runtime.CamelRuntime.doStart(CamelRuntime.java:90)
	at org.apache.camel.support.service.ServiceSupport.start(ServiceSupport.java:86)
	at io.quarkus.camel.core.runtime.CamelTemplate.start(CamelTemplate.java:35)
	at io.quarkus.deployment.steps.CamelInitProcessor$createRuntimeInitTask6.deploy(Unknown Source)
	at io.quarkus.runner.ApplicationImpl1.doStart(Unknown Source)
	at io.quarkus.runtime.Application.start(Application.java:93)
	at io.quarkus.runtime.Application.run(Application.java:198)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
Caused by: java.lang.ClassNotFoundException: com.github.benmanes.caffeine.cache.SILMS
	at com.oracle.svm.core.hub.ClassForNameSupport.forName(ClassForNameSupport.java:51)
	at java.lang.ClassLoader.loadClass(Target_java_lang_ClassLoader.java:127)
```

--- a/extensions/camel/camel-core/deployment/pom.xml
+++ b/extensions/camel/camel-core/deployment/pom.xml
@@ -1,56 +1,50 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project xmlns="http://maven.apache.org/POM/4.0.0"
-  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
-  <parent>
-    <artifactId>quarkus-camel-core-parent</artifactId>
-    <groupId>io.quarkus</groupId>
-    <version>999-SNAPSHOT</version>
-    <relativePath>../</relativePath>
-  </parent>
-  <modelVersion>4.0.0</modelVersion>
-
-  <artifactId>quarkus-camel-core-deployment</artifactId>
-  <name>Quarkus - Camel - Core - Deployment</name>
-
-  <dependencies>
-
-    <!-- quarkus -->
-    <dependency>
-      <groupId>io.quarkus</groupId>
-      <artifactId>quarkus-core-deployment</artifactId>
-    </dependency>
-    <dependency>
-      <groupId>io.quarkus</groupId>
-      <artifactId>quarkus-arc-deployment</artifactId>
-    </dependency>
-
-    <!-- camel -->
-    <dependency>
-      <groupId>io.quarkus</groupId>
-      <artifactId>quarkus-camel-core</artifactId>
-    </dependency>
-    <dependency>
-      <groupId>io.quarkus</groupId>
-      <artifactId>quarkus-jaxb-deployment</artifactId>
-    </dependency>
-
-  </dependencies>
-
-  <build>
-    <plugins>
-      <plugin>
-        <artifactId>maven-compiler-plugin</artifactId>
-        <configuration>
-          <annotationProcessorPaths>
-            <path>
-              <groupId>io.quarkus</groupId>
-              <artifactId>quarkus-extension-processor</artifactId>
-              <version>${project.version}</version>
-            </path>
-          </annotationProcessorPaths>
-        </configuration>
-      </plugin>
-    </plugins>
-  </build>
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>quarkus-camel-core-parent</artifactId>
+        <groupId>io.quarkus</groupId>
+        <version>999-SNAPSHOT</version>
+        <relativePath>../</relativePath>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>quarkus-camel-core-deployment</artifactId>
+    <name>Quarkus - Camel - Core - Deployment</name>
+    <dependencies>
+        <!-- quarkus -->
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-core-deployment</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-arc-deployment</artifactId>
+        </dependency>
+        <!-- camel -->
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-camel-core</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-jaxb-deployment</artifactId>
+        </dependency>
+    </dependencies>
+    <build>
+        <plugins>
+            <plugin>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <annotationProcessorPaths>
+                        <path>
+                            <groupId>io.quarkus</groupId>
+                            <artifactId>quarkus-extension-processor</artifactId>
+                            <version>${project.version}</version>
+                        </path>
+                    </annotationProcessorPaths>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
 </project>

--- a/extensions/camel/camel-core/deployment/pom.xml
+++ b/extensions/camel/camel-core/deployment/pom.xml
@@ -30,6 +30,10 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-jaxb-deployment</artifactId>
         </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-caffeine-deployment</artifactId>
+        </dependency>
     </dependencies>
     <build>
         <plugins>

--- a/extensions/camel/camel-core/runtime/pom.xml
+++ b/extensions/camel/camel-core/runtime/pom.xml
@@ -48,6 +48,10 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-jaxb</artifactId>
         </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-caffeine</artifactId>
+        </dependency>
     </dependencies>
 
     <build>

--- a/extensions/camel/camel-core/runtime/pom.xml
+++ b/extensions/camel/camel-core/runtime/pom.xml
@@ -44,7 +44,10 @@
                 </exclusion>
             </exclusions>
         </dependency>
-
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-jaxb</artifactId>
+        </dependency>
     </dependencies>
 
     <build>

--- a/integration-tests/camel-aws-s3/pom.xml
+++ b/integration-tests/camel-aws-s3/pom.xml
@@ -38,11 +38,6 @@
         </dependency>
 
         <dependency>
-            <groupId>io.quarkus</groupId>
-            <artifactId>quarkus-caffeine</artifactId>
-        </dependency>
-
-        <dependency>
             <groupId>org.jboss.slf4j</groupId>
             <artifactId>slf4j-jboss-logging</artifactId>
             <version>1.1.0.Final</version>

